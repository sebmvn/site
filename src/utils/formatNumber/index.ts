export const formatPriceNotation = (value: string) => {
  value = value?.replace('.', '').replace(',', '').replace(/\D/g, '')

  if (isNaN(parseFloat(value))) {
    return ''
  }

  const options = { minimumFractionDigits: 2 }
  const result = new Intl.NumberFormat('pt-BR', options).format(
    parseFloat(value) / 100
  )

  return result
}

export const formatIntegerNotation = (value: string) => {
  return value?.replace('.', '').replace(',', '').replace(/\D/g, '')
}

export const toDecimal = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseFloat(formattedValue!) / 100
}

export const toInteger = (value: string) => {
  const formattedValue = formatIntegerNotation(value)
  return parseInt(formattedValue!)
}
