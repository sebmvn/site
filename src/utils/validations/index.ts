import { CalendarType } from 'hooks/use-calc'
import Joi from 'joi'

const fieldsValidations = {
  initialValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  monthlyValue: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),

  interestRate: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  period: Joi.string().required().messages({
    'any.required': 'Campo obrigatório',
    'string.empty': 'Campo obrigatório',
    'string.min': 'tamanho minimo de 1 digito',
    'string.base': 'o tipo do dado precisa ser textual'
  }),
  interestRateCalendar: Joi.string().valid('monthly', 'yearly').required(),
  periodCalendar: Joi.string().valid('monthly', 'yearly').required()
}

export type FieldErrors = {
  [key: string]: string
}

const getFieldErrors = (objError: Joi.ValidationResult) => {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type compoundInterestCalculatorFields = {
  initialValue: string
  monthlyValue: string
  interestRate: string
  period: string
  interestRateCalendar: CalendarType
  periodCalendar: CalendarType
}
export const compoundInterestCalculatorValidate = (
  values: compoundInterestCalculatorFields
) => {
  const schema = Joi.object(fieldsValidations)

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}
