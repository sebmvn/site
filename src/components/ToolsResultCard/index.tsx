import { Card } from 'components/Card'
import * as S from './styles'

export type ToolsResultCardProps = {
  title: string
  value: string
}

const ToolsResultCard = ({ title, value }: ToolsResultCardProps) => (
  <S.Wrapper>
    <Card>
      <S.Title>{title}</S.Title>
      <S.Value>{value}</S.Value>
    </Card>
  </S.Wrapper>
)

export default ToolsResultCard
