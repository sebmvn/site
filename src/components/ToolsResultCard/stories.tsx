import { Story, Meta } from '@storybook/react/types-6-0'
import ToolsResultCard, { ToolsResultCardProps } from '.'

export default {
  title: 'ToolsResultCard',
  component: ToolsResultCard,
  args: {
    title: 'Valor total final',
    value: 'R$ 30.999,13'
  }
} as Meta

export const Default: Story<ToolsResultCardProps> = (args) => (
  <ToolsResultCard {...args} />
)
