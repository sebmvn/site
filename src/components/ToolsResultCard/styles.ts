import styled, { css } from 'styled-components'
import { Card } from 'components/Card'

export const Wrapper = styled.article`
  ${Card} {
    align-items: center;
    text-align: center;
  }
`

export const Title = styled.h4`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.normal};
  `}
`

export const Value = styled.data`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.black};
  `}
`
