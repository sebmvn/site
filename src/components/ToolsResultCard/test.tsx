import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test-utils'

import ToolsResultCard from '.'

describe('<ResultCard />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <ToolsResultCard title="ToolsResultCard" value="R$ 30.999,13" />
    )

    expect(
      screen.getByRole('heading', { name: /ToolsResultCard/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
