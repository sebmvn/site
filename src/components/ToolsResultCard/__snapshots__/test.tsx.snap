// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultCard /> should render the heading 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: #FFFFFF;
  padding: 3.2rem 2.4rem;
  border: 1px solid #F4F4F4;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px #1932413F;
}

.c0 .c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c3 {
  color: #193241;
  font-size: 1.4rem;
  font-weight: 400;
}

.c4 {
  font-size: 2.0rem;
  font-weight: 900;
}

<article
  class="c0"
>
  <div
    class="c1 c2"
  >
    <h4
      class="c3"
    >
      ToolsResultCard
    </h4>
    <data
      class="c4"
    >
      R$ 30.999,13
    </data>
  </div>
</article>
`;
